Chaincase.Common/Global.cs

```diff
@@ -365,18 +356,15 @@ public async Task OnResuming()
                     Logger.LogDebug($"{nameof(Global)}.OnResuming(): Entered critical section");
 
                     // don't ever cancel Init. use an ephemeral token
-                    await WaitForInitializationCompletedAsync(new CancellationToken());
+                    await InitializationCompleted.Task;
 
                     var userAgent = Constants.UserAgents.RandomElement();
                     var connectionParameters = new NodeConnectionParameters { UserAgent = userAgent };
                     var addrManTask = InitializeAddressManagerBehaviorAsync();
                     AddressManagerBehavior addressManagerBehavior = await addrManTask.ConfigureAwait(false);
                     connectionParameters.TemplateBehaviors.Add(addressManagerBehavior);
-
-                    if (_torManager?.State != TorState.Started && _torManager.State != TorState.Connected)
-                    {
-                        await _torManager.StartAsync(false, DataDir);
-                    }
+                    await _torManager.StartAsync(ResumeCts.Token);
```

Comment:
Because it's written in c we don't have full control over the tor process. The conditional was added from the reference implementation @ OnionBrowser. Why remove?